eva/optimizer/optimizer_tasks.py

```diff
@@ -191,15 +189,6 @@ def execute(self):
                 ApplyRule(rule, self.root_expr, self.optimizer_context, self.explore)
             )
 
-            # explore the input group if necessary
```

Comment:
Can we rather comment this code? I don't remember why we need this, but most likely was part of the state machine.

Reply:
Yes, added it back and commented it.